### ISSUE TITLE
Invoke SetIdleHint() on lock/unlock

### ISSUE THREAD

**Issue by andreyv** [the-cavalry/light-locker#52]
I'm looking to make systemd suspend the system when all user sessions are idle. `man logind.conf` says this about the `IdleAction` parameter:

> Note that this requires that user sessions correctly report the idle status to the system. The system will execute the action after all sessions report that they are idle, no idle inhibitor lock is active, and subsequently, the time configured with IdleActionSec= (see below) has expired.

It seems that the moment of locking/unlocking the session in light-locker is a good candidate to update this idle status. This document somewhat describes this topic: https://wiki.freedesktop.org/www/Software/systemd/writing-desktop-environments/

**Comment by ochosi** [comment]
True, light-locker would be one possible place to implement this.
@cavalier38: Or would it be better if this went directly into lightdm or xfce4-session?

**Comment by andreyv** [comment]
Actually after opening the issue I had some second thoughts about whether light-locker is the right place for this, so I'm glad you pointed it out :)

**Comment by cavalier38** [comment]
This can be added to light-locker, as light-locker does bring a x11 screensaver to dbus api, it can also add this as another x11 to dbus bridge.

If light-locker is the right place all depends on the locker/session-manage/power-manager sepperation which is pretty unclear currently.

As we already include extras in light-locker to fix some problems, we could add this to light-locker with a runtime option like all the other things. So it can be disabled if another part handles this.

**Comment by chairrrr** [comment]
I would like this too, if that makes a difference. I'm not running a DE, just a WM, so no session manager or power manager. I can't seem to find a suitable lightweight power manager and really sleeping on idle is the only thing that I still need.

**Comment by cavalier38** [comment]
If this goes into light-locker. What should we consider idle?
- The moment the screensaver activates?
- The moment the session is locked? (It could switch to the lock screen at that time, are locked sessions idle?)
- A configurable time after the screensaver acticates?

**Comment by andreyv** [comment]
The session is idle when it is not being used. The screensaver also activates when the session is not being used. It seems to me that joining these events would be fair.

**Comment by andreyv** [comment]
Just for the record, it seems that this was implemented in commit f347c44d1743de6e9f1f0550434c0ac79f7dadc7.

However, with version 1.7.0 and with running `light-locker --idle-hint` the session is still not being set idle. I may be able to provide debug logs later.
